provisioner/ansible-local/provisioner.go

```diff
@@ -322,5 +325,20 @@ func (p *Provisioner) uploadDir(ui packer.Ui, comm packer.Communicator, dst, src
 	if src[len(src)-1] != '/' {
 		src = src + "/"
 	}
-	return comm.UploadDir(dst, src, nil)
+
+	if err := comm.UploadDir(dst, src, nil); err != nil {
+		return err
+	}
+
+	for _, element := range p.config.Exclude {
+		cmd := &packer.RemoteCmd{
+			Command: fmt.Sprintf("rm -rf '%s'", filepath.Join(dst, element)),
```

Comment:
This looks dangerous. If there are files we don't want to upload we should just not upload them, not upload and then delete things. I think this could have the effect of deleting things I didn't expect or leaking secrets into a box that don't get cleaned up properly.